module/VuFind/tests/unit-tests/src/VuFindTest/Marc/MarcCollectionFileTest.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * MarcCollectionFile Test Class
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+namespace VuFindTest\Marc;
+
+use VuFind\Marc\MarcCollectionFile;
+
+/**
+ * MarcCollectionFile Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class MarcCollectionFileTest extends \PHPUnit\Framework\TestCase
+{
+    use \VuFindTest\Feature\FixtureTrait;
+
+    /**
+     * Return collection fixtures for the tests
+     *
+     * @return array
+     */
+    public function collectionFixtures(): array
+    {
+        return [
+            ['marc/marc_collection.xml'],
```

Comment:
Might it be worth adding a third fixture containing non-namespaced XML? For example, if somebody harvests MARCXML using VuFind's OAI-PMH tool, it's possible they could end up with namespaced `<record>` elements contained within a non-namespaced `<collection>` container (if they use the default behavior for grouping batches of harvested records). Might be good to ensure that this works appropriately. I'm happy to create the fixture and add the test, but I thought I'd see if you found the idea problematic before trying it.

Reply:
I added a third fixture that contains a non-namespaced collection element and also a record from another namespace and fixed the collection reader to actually look for a correct namespace. It's still a bit lenient as it also allows records without a namespace, but at least it will now skip records with a bad one.

Reply:
I think that makes sense. Only one more question: should we emit a notice or warning if we encounter a bad namespace? I'm imagining a scenario where somebody builds XML by hand and makes a typo in the namespace URI, then can't figure out why the record won't load. Another scenario: since the namespace URI contains http:// and there's increasing pressure to move to https://, I could imagine a global search-and-replace going too far and confusing things. :-)

Reply:
I'd be inclined to say "no" at least without a verbose flag, since it would be analogous to encountering e.g. an `<item>` element instead of `<record>`. If you believe such flag would be useful, I guess it would be ok to do something like call `trigger_error()` with E_WARNING level if a verbose flag is set. What do you think?

WRT http vs. https, as far as I know XML namespace URLs should be considered opaque identifiers, and using https would mean that it's not the same schema as with http. Thus I wouldn't accept https in this case.

Reply:
A verbose flag might be helpful, but if it's a nuisance to implement, I don't think it's essential on day one. (Though if you do want to implement it, I'm also thinking it might make sense to switch it on when calling this from the linter code, which would be a good way to demonstrate/test the functionality).

I absolutely agree about XML namespace URLs -- changing http to https would NOT be correct. It just seems like a naive mistake that somebody might make. :-)

Reply:
I don't particularly want to implement it, so I'd be happy to defer it to a later date. :)

I can actually imagine a situation where someone comes up with the great idea that they can interleave normal MARC records and "enhanced" format by using https for the enhanced format's schema. Or maybe it's just MARCXML 2.0. In any case, I still don't think it's a good idea to support something that's not done right.

Reply:
I think I'll add the verbose support after all since it has a minor effect on the constructor.

Reply:
Done as a separate commit. I decided to implement it as a callback so that anyone using the class can display the messages in a way they want, and did that for the deletes command.